src/__tests__/index.test.ts

```diff
@@ -731,54 +732,51 @@ test('closing before ever connecting', (done) => {
   );
 });
 
-// // re-add once we add polling
-// test.skip('falling back to polling', (done) => {
-//   const onUnrecoverableError = jest.fn<void, [Error]>();
-//   const client = new Client();
-//   client.setUnrecoverErrorHandler(onUnrecoverableError);
-
-//   const maxConnectRetries = 1;
-//   const open = jest.fn();
-
-//   client.setDebugFunc((log) => {
-//     if (log.type === 'breadcrumb' && log.message === 'falling back to polling') {
-//       // eslint-disable-next-line
-//       const data = log.data as any;
-
-//       expect(data.connectTries).toEqual(maxConnectRetries + 1);
-//       expect(data.error).toBeDefined();
-//       expect(data.wsReadyState).toBeUndefined();
-
-//       // eslint-disable-next-line
-//       // @ts-ignore need to reach in and grab some private fields real quick...
-//       const { urlOptions, polling } = client.connectOptions;
-
-//       expect(urlOptions.host).toEqual('gp-v2.herokuapp.com');
-//       expect(polling).toBe(true);
-
-//       expect(open).not.toHaveBeenCalled();
-
-//       expect(onUnrecoverableError).toHaveBeenCalledTimes(0);
-//       done();
-//     }
-//   });
-
-//   client.open(
-//     {
-//       fetchConnectionMetadata: () => Promise.resolve({
-//         connectionMetadata: {
-//           token: 'bad token',
-//           gurl: '',
-//           conmanURL: '',
-//         },
-//         result: FetchConnectionMetadataResult.Ok,
-//       }),
-//       WebSocketClass: WebSocket,
-//       timeout: 0,
-//     },
-//     open,
-//   );
-// });
+test('fallback to polling', (done) => {
+  const onUnrecoverableError = jest.fn<void, [Error]>();
+  const client = new Client();
+  client.setUnrecoverableErrorHandler(onUnrecoverableError);
+
+  class WebsocketThatNeverConnects {
+    static OPEN = 1;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  expect(() => getWebSocketClass(WebsocketThatNeverConnects)).not.toThrow();
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  expect(getWebSocketClass(WebsocketThatNeverConnects)).toEqual(WebsocketThatNeverConnects);
+
+  let didLogFallback = false;
+  client.setDebugFunc((log) => {
+    if (log.type === 'breadcrumb' && log.message === 'polling fallback') {
+      didLogFallback = true;
+    }
+  });
+
+  client.open(
+    {
+      timeout: 2000,
+      fetchConnectionMetadata: () =>
+        Promise.resolve({
+          ...genConnectionMetadata(),
+          error: null,
+        }),
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      WebSocketClass: WebsocketThatNeverConnects,
+      context: null,
+    },
+    ({ channel }) => {
+      expect(channel).toBeTruthy();
+      expect(didLogFallback).toBeTruthy();
```

Comment:
nit: maybe we can be more specific here? e.g. 
```suggestion
    ({ channel, error }) => {
      expect(error).toBeNull();
      expect(channel).not.toBeNull();
      expect(didLogFallback).toBe(true);
```